repair/repair_task.hh

```diff
@@ -23,6 +23,30 @@ public:
     virtual future<> run() override = 0;
 };
 
+class root_repair_task_impl : public repair_task_impl {
+public:
+    root_repair_task_impl(task_manager::module_ptr module, task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "")
+        : repair_task_impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity)) {}
+protected:
+    repair_uniq_id get_repair_uniq_id() {
+        return repair_uniq_id{
+            .id = _status.sequence_number,
+            .uuid = _status.id
+        };
+    }
+    future<> run() override = 0;
+};
+
+class user_requested_repair_task_impl : public root_repair_task_impl {
+private:
+    std::unordered_map<sstring, sstring> _options_map;
+public:
+    user_requested_repair_task_impl(task_manager::module_ptr module, task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", std::unordered_map<sstring, sstring> options_map = {})
+        : root_repair_task_impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity)), _options_map(std::move(options_map)) {}
+protected:
+    future<> run() override;
```

Comment:
1. Let's add a TODO for implementing abort for user-initiated repair.
2. How are the repair tasks aborted today? Should / can we integrate that with the task_manager abort source?

Reply:
2. When repair tasks are aborted, the `abort` flag is set to `true`. The flag's value is checked before some operation and an exception is thrown if the flag was set.
Abort source of `task_manager` is the one that is activated on shutdown. It is something different from that abort, but there is a analogical `shutdown` flag in repairs and I believe this one may be integrated. I will take a closer look on that.

Reply:
as we discussed verbally, let's stick to the current abort design while keeping the repair tasks non-abortable from the task_manager api.
Then later we can transition to using an abort_source per task so we can abort individual tasks, and on top of subscribe to the higher level abort_source and when call on abort, individually abort all outstanding tasks and wait on them to resolve and unregister.